odpf/assets/v1beta2/metric.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package odpf.assets.v1beta2;
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/odpf/proton/assets/v1beta2;assetsv1beta2";
+option java_outer_classname = "MetricProto";
+option java_package = "io.odpf.assets";
+
+// A metric is a timeseries aggregation over a table that supports zero or more dimensions.
+message Metric {
+  // Namespace of the metric. Something like `{project}-{model}` for dbt and
+  // `schema` for Cube.
+  string namespace = 1;
+
+  // The field being used to calculate a metric.
+  string field_name = 2;
+
+  // Type of the evaluated metric. ex: count_distinct, average etc.
+  string measure_type = 3;
+
+  // Optional: The SQL representation of the query with filters and aggregations.
+  string sql = 4;
```

Comment:
@sudo-suhas lets call it query instead of SQL. Because the query interface could be anything and not just SQL.